kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kroxylicious/KroxyliciousResource.java

```diff
@@ -36,29 +34,36 @@ public String getKind() {
         return resourceClass.getSimpleName();
     }
 
-    @Override
-    public T get(String namespace, String name) {
-        return kubeClient().inNamespace(namespace).withName(name).get();
-    }
-
     @Override
     public void create(T resource) {
-        kubeClient().inNamespace(resource.getMetadata().getNamespace()).resource(resource).create();
+        getClient().inNamespace(resource.getMetadata().getNamespace()).resource(resource).create();
     }
 
     @Override
     public void delete(T resource) {
-        kubeClient().inNamespace(resource.getMetadata().getNamespace()).withName(
+        getClient().inNamespace(resource.getMetadata().getNamespace()).withName(
                 resource.getMetadata().getName()).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
     }
 
     @Override
-    public void update(T resource) {
-        kubeClient().inNamespace(resource.getMetadata().getNamespace()).resource(resource).update();
+    public void replace(T resource, Consumer<T> editor) {
+        T toBeUpdated = getClient().inNamespace(resource.getMetadata().getNamespace()).withName(resource.getMetadata().getName()).get();
+        editor.accept(toBeUpdated);
+        this.update(toBeUpdated);
     }
 
     @Override
-    public boolean waitForReadiness(T resource) {
+    public boolean isReady(T resource) {
```

Comment:
The implementation is `isReady` and `isDeleted` is surprising.   I know we've spoken about it before.  The other methods are dependent on server state.    At least add a comment explaining why this methods are implemented as they are.